src/widget/input_box.h

```diff
@@ -0,0 +1,18 @@
+#ifndef WIDGET_INPUT_BOX_H
+#define WIDGET_INPUT_BOX_H
+
+#include "input/mouse.h"
+
+#define INPUT_BOX_BLOCK_SIZE 16
+
+typedef struct {
+    int x;
+    int y;
+    int w;
```

Comment:
Suggestion: call `w` and `h`, `width_blocks` and `height_blocks`, to make it clear that the unit is blocks instead of pixels.